tests/mock.test.ts

```diff
@@ -40,7 +40,10 @@ describe("Mock", () => {
 			},
 		};
 		expect(mock(request, options)).toBe(true);
-		expect(mock(request, options)).toBe(undefined);
+
+		// for some reason the expect function fails to infer the return type correctly and only infers
+		// it as boolean so we specify the type of expect explicitly as the return type of the function
+		expect<ReturnType<typeof mock>>(mock(request, options)).toBe(undefined);
```

Comment:
this seems like a bug within the types of bun:test, I will investigate it more and eventually create an issue on their repo